ui/src/app/views/app/component.ts

```diff
@@ -102,4 +105,30 @@ export class AppComponent implements OnInit {
     }
   }
 
+  /**
+   * Setting the Google Analytics ID here.
+   * This first injects Google's gtag script via iife, then secondarily defines
+   * the global gtag function.
+   */
+  private setGTagManager() {
+    const s = this.doc.createElement('script');
+    s.type = 'text/javascript';
+    s.innerHTML =
+      '(function(w,d,s,l,i){' +
```

Comment:
What are the various variables here? (Just not really sure what everything is. Is this copied from somewhere?)

Reply:
This function is invoked with the variables provided on line 126. This is mostly copied code from the angular boards, but it's using a javascript construct called "iife" which defines a function, then immediately invokes it. 

In this obscure block of code, the iife is inserting a script tag into the head of the document that looks like this:
```
<script async src="https://www.googletagmanager.com/gtag/js?id=[ga id variable here]"></script>
```